app/Actions/Photo/Create.php

```diff
@@ -184,7 +185,7 @@ protected function findLivePartner(
 
 			return $livePartner;
 		} catch (IllegalOrderOfOperationException $e) {
-			assert(false, new \AssertionError('IllegalOrderOfOperationException must not be thrown', $e));
+			throw new \AssertionError('IllegalOrderOfOperationException must not be thrown', $e->getCode(), $e);
```

Comment:
Php Stan is complaining that assert(false, xxx) will always throw and should not be used.
Maybe we should consider instead a proper utility function throwIfDebug() or considering disabling that rule.

I am fine with either solutions.

Reply:
Using a utility function like `throwIfDebug()` is not the solution, because this will always mean a real function call which is not optimized away.

Since, PHP 7 `assert` is not an ordinary function call anymore, but a language keyword like `if`, `while`, `public`, etc. I am a little bit disappointed about PhpStan. The complain that `assert(false, xxx)` will always throw is simply wrong. It won't throw, if it is optimized away. We are not talking about an environment setting like the one provided by Laravel `.env`, i.e. a configuration option like `ENABLE_DEBUG`, but a language feature built into the Zend engine which is used by opcode optimizer.

Are we sure that PhpStan knows that we are only targeting PHP 8 and above? The complain would be correct for PHP 6.x and below. Otherwise, really disappointing indeed ...

The whole idea of wrapping that thing into an `assert` was to get rid of annoying messages that an exception is neither handled nor caught. Replacing the exception by another non-caught, non-handled exception does not help.

Reply:
Fixed in https://github.com/LycheeOrg/Lychee/pull/1336/commits/f5cd4433a97aef5706f3d661e5c9578e4e7e9938. See https://github.com/LycheeOrg/Lychee/pull/1336#discussion_r894998507 for the full story.